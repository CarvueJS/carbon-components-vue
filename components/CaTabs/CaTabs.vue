<template>
  <nav data-tabs class="bx--tabs" role="navigation">
    <div class="bx--tabs-trigger" tabindex="0">
      <a href="javascript:void(0)" class="bx--tabs-trigger-text" tabindex="-1"></a>
      <svg width="10" height="5" viewBox="0 0 10 5" fill-rule="evenodd">
        <path d="M10 0L5 5 0 0z"></path>
      </svg>
    </div>
    <ul class="bx--tabs__nav bx--tabs__nav--hidden" role="tablist">
      <li class="bx--tabs__nav-item bx--tabs__nav-item--selected" data-target=".tab-1" role="presentation">
        <a id="tab-link-1" class="bx--tabs__nav-link" href="javascript:void(0)" role="tab" aria-controls="tab-panel-1" aria-selected="true">Tab label 1</a>
      </li>
      <li class="bx--tabs__nav-item" data-target=".tab-2" role="presentation">
        <a id="tab-link-2" class="bx--tabs__nav-link" href="javascript:void(0)" role="tab" aria-controls="tab-panel-2" aria-selected="false">Tab label 2</a>
      </li>
      <li class="bx--tabs__nav-item" data-target=".tab-3" role="presentation">
        <a id="tab-link-3" class="bx--tabs__nav-link" href="javascript:void(0)" role="tab" aria-controls="tab-panel-3" aria-selected="false">Tab label 3</a>
      </li>
      <li class="bx--tabs__nav-item" data-target=".tab-4" role="presentation">
        <a id="tab-link-4" class="bx--tabs__nav-link" href="javascript:void(0)" role="tab" aria-controls="tab-panel-4" aria-selected="false">Tab label 4</a>
      </li>
      <li class="bx--tabs__nav-item" data-target=".tab-5" role="presentation">
        <a id="tab-link-5" class="bx--tabs__nav-link" href="javascript:void(0)" role="tab" aria-controls="tab-panel-5" aria-selected="false">Tab label 5</a>
      </li>
    </ul>
  </nav>
  <!-- The markup below is for demonstration purposes only -->
  <div style="padding: 1rem;">
    <div id="tab-panel-1" class="tab-1" role="tabpanel" aria-labelledby="tab-link-1" aria-hidden="false">
      <p>Content for first tab goes here.</p>
    </div>
    <div id="tab-panel-2" class="tab-2" role="tabpanel" aria-labelledby="tab-link-2" aria-hidden="true" hidden>
      <p>Content for second tab goes here.</p>
    </div>
    <div id="tab-panel-3" class="tab-3" role="tabpanel" aria-labelledby="tab-link-3" aria-hidden="true" hidden>
      <p>Content for third tab goes here.</p>
    </div>
    <div id="tab-panel-4" class="tab-4" role="tabpanel" aria-labelledby="tab-link-4" aria-hidden="true" hidden>
      <p>Content for fourth tab goes here.</p>
    </div>
    <div id="tab-panel-5" class="tab-5" role="tabpanel" aria-labelledby="tab-link-5" aria-hidden="true" hidden>
      <p>Content for fifth tab goes here.</p>
    </div>
  </div>
</template>

<script>
  export default {
    name: 'ca-tabs'
  }
</script>

<style></style>
