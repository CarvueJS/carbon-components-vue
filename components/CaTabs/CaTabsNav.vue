<template>
  <li class="bx--tabs__nav-item" :class="dynamicClass" :data-target="tabID" role="presentation">
    <a class="bx--tabs__nav-link" href="javascript:void(0)" role="tab" :aria-controls="controls" :aria-selected="selected">
      <slot></slot>
    </a>
  </li>
</template>

<script>
export default {
  name: 'ca-tabs-nav',
  props: {
    controls: {
      type: String,
      default: '',
    },
    selected: {
      type: Boolean,
      default: false,
    },
    tabID: {
      type: String,
      default: '',
    },
  },
  computed: {
    dynamicClass() {
      return {
        'bx--tabs__nav-item--selected': this.selected,
      };
    },
  },
};
</script>

<style>

</style>
