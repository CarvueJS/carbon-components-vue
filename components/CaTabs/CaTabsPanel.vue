<template>
  <div :id="PanelID" :class="tabID" role="tabpanel" :aria-labelledby="labelledby" :aria-hidden="ariaHidden" :hidden="hidden">
    <slot></slot>
  </div>
</template>

<script>
  export default {
    name: 'ca-tabs-panel',
    props: {
      PanelID: {
        type: String,
        default: '',
      },
      labelledby: {
        type: String,
        default: '',
      },
      hidden: {
        type: Boolean,
        default: true,
      },
      ariaHidden: {
        type: Boolean,
        default: true,
      },
      tabID: {
        type: String,
        default: '',
      },
    },
  };
</script>


