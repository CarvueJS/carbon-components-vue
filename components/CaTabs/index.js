import CaTabs from './CaTabs.vue';

CaTabs.install = Vue => {
  Vue.component(CaTabs.name, CaTabs)
}

export default CaTabs
