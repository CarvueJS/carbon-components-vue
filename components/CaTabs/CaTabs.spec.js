import { shallow } from 'vue-test-utils';
import CaTabs from './CaTabs.vue';

describe('CaTabs', () => {
  test('is a Vue instance', () => {
    const wrapper = shallow(CaTabs);
    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  test('Mathes snapshot', () => {
    const wrapper = shallow(CaTabs);
    expect(wrapper.html()).toMatchSnapshot()
  });
})
