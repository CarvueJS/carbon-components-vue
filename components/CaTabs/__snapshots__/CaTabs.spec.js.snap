// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CaTabs Mathes snapshot 1`] = `
<div>
  <nav data-tabs="" role="navigation" class="bx--tabs">
    <div tabindex="0" class="bx--tabs-trigger">
      <a href="javascript:void(0)" tabindex="-1" class="bx--tabs-trigger-text"></a>
      <svg width="10" height="5" viewBox="0 0 10 5" fill-rule="evenodd">
        <path d="M10 0L5 5 0 0z"></path>
      </svg>
    </div>
    <ul role="tablist" class="bx--tabs__nav bx--tabs__nav--hidden"></ul>
  </nav>
  <div style="padding: 1rem;"></div>
</div>
`;
